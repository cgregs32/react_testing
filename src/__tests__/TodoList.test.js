import React from 'react';
import { shallow, mount } from 'enzyme';
import ConnectedComponent, { TodoList } from '../TodoList';
import toJson from 'enzyme-to-json';
import { reduxHelper } from '../reduxHelper';

const initialState = { items: [] }

describe(<TodoList />, () => {

  describe('render', () => {
    let component;

    beforeEach( () => {
      const tree = reduxHelper(ConnectedComponent, initialState).component
      component = mount(tree);
    });

    it('renders w/o crashing', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

  });

  describe('functionality', () => {
    let component;

    beforeEach( () => {
      component = mount(<TodoList />)
      let input = component.find('input');
      input.simulate('focus');
      inpit.simulate(
        'change',
        { target: { name: 'name', value: 'hello' }}
      )
    });

    it('submits the form', () => {
      let component = shallow(<TodoList items={[]} />)
      expect(component.state('items').length).toEQual(0)
      component.find('form').simulate('submit');
      expect(component.state('items').length).toEqual(1);
    })

    it('updates state on change', () => {
      expect(component.state('name')).toEqual('hello')
    });

  });

});
