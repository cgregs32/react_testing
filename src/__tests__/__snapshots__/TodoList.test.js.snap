// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] render renders w/o crashing 1`] = `
<div>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Add Item"
      required={true}
      value=""
    />
  </form>
  <hr />
  <ul />
</div>
`;
